octopoes/models/ooi/web.py

```diff
@@ -194,3 +197,95 @@ def format_reference_human_readable(cls, reference: Reference) -> str:
         address = t.resource.website.ip_service.ip_port.address.address
 
         return f"{t.key} @ {web_url} @ {address} contains {str(reference.tokenized.hostname.name)}"
+
+
+class HTTPCookie(OOI):
+    # https://datatracker.ietf.org/doc/html/rfc6265
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies
+    object_type: Literal["HTTPCookie"] = "HTTPCookie"
+
+    _natural_key_attrs = [
+        "name",
+        "domain",
+        "path",
+    ]  # does not include network, as there is not concept of a network only hostnames in the browsers cookie handling.
+
+    # httpheader: Reference(ReferenceField(HTTPHeaderURL, max_inherit_scan_level=4))  # ideally we should keep track of which header was parsed into this cookie object
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3
+    name: str
+    value: str
+    expirytime: Optional[datetime]
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p6
+    domain: Reference = ReferenceField(Hostname, max_inherit_scan_level=4)
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p7
+    path: str = "/"
+    creationtime: datetime
+    # last-access-time: # not used in openkat context
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p3
+    persistent: bool = False
+
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p6
+    hostonly: bool = False
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p8
+    secureonly: bool = False
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p9
+    httponly: bool = False
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
+    samesite: bool = False
+    maxage: Optional[int]
+
+    @classmethod
+    def format_reference_human_readable(cls, reference: Reference) -> str:
+        tokenized = reference.tokenized
+        return f"{tokenized.name} @ {tokenized.domain}"
+
+    @classmethod
+    def fromstring(cls, responsedomain, cookie) -> Iterator[OOI]:
```

Comment:
A few things:
- `Iterator` is missing from imports
- Some variable names (in both signature as body) don't adhere to the [PEP8 naming conventions](https://pep8.org/#naming-conventions)